### PR TITLE
Mandatory tests part 5

### PR DESCRIPTION
**Pull request by domachine**
This PR includes more mandatory tests:

- 6.1.27.1
- 6.1.27.2
- 6.1.27.3
- 6.1.27.4
- 6.1.27.5
- 6.1.27.6
- 6.1.27.7
- 6.1.27.8
- 6.1.27.10

### CODE REVIEW

**Review by wrtlbrmpft — APPROVED** [review]
Looks good to me.

File: app/lib/shared/Core/entities/DocumentEntity/mandatoryTest_6_1_27_6.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/**
+ * @param {any} doc
+ */
+export default function (doc) {
+  /** @type {Array<{ message: string; instancePath: string }>} */
+  const errors = []
+  let isValid = true
+
+  const checkedDocumentCategories = new Set(['security_advisory', 'vex'])
```
Comment (tschmidtb51) [inline app/lib/shared/Core/entities/DocumentEntity/mandatoryTest_6_1_27_6.js:9]: @domachine: This should only test on `security_advisory` as `vex` has a special test for that:
```suggestion
  const checkedDocumentCategories = new Set(['security_advisory'])
```

File: app/lib/shared/Core/entities/DocumentEntity/mandatoryTest_6_1_27_7.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  const checkedDocumentCategories = new Set(['vex'])
+
+  if (!checkedDocumentCategories.has(doc.document?.category))
+    return { errors, isValid }
+
+  /** @type {unknown} */
+  const vulnerabilities = doc.vulnerabilities
+  if (Array.isArray(vulnerabilities)) {
+    vulnerabilities.forEach((vulnerability, vulnerabilityIndex) => {
+      if (!vulnerability.product_status) return
```
Comment (tschmidtb51) [inline app/lib/shared/Core/entities/DocumentEntity/mandatoryTest_6_1_27_7.js:18]: As 6.1.27.6 (see previous review comment) does not longer covers the part of checking whether the product status is there, it must be implemented here.

File: app/tests/Core/documentTests.js [hunk 3]
```diff
@@ -1647,6 +1648,14 @@ export default [
           product_status: {
             fixed: ['CSAFPID-0001'],
           },
+          ...(doc === minimalVexDoc
+            ? {
+                id: {
+                  system_name: 'My system',
+                  text: 'My system test',
```
Comment (tschmidtb51) [inline app/tests/Core/documentTests.js:1655]: Just to make the usage clear => link to example from spec.
```suggestion
                  system_name: 'GitHub Issue',
                  text: 'oasis-tcs/csaf#210',

File: app/tests/Core/documentTests.js [hunk 4]
```diff
@@ -1686,6 +1695,10 @@ export default [
               text: 'My note',
             },
           ],
+          id: {
+            system_name: 'My system',
+            text: 'My system test',
```
Comment (tschmidtb51) [inline app/tests/Core/documentTests.js:1700]: Just to make the usage clear => link to example from spec.
```suggestion
            system_name: 'GitHub Issue',
            text: 'oasis-tcs/csaf#210',

**Review by tschmidtb51 — CHANGES_REQUESTED** [review]
@domachine: Overall a solid implementation (and shows some room from improvement in the spec...).
Minor comments regarding examples and wording.

As it is the reference implementation, I would like to be strict on the separation of 6.1.27.6 and .7 (even though your idea of checking that in .6 was great => as I said: room for improvement in the spec).

**Comment by tschmidtb51** [comment]
@domachine As discussed, please also do 3e0dea6 in a separate PR (new review required as it is just dependency update work as in AP 1).

**Comment by domachine** [comment]
Alright, I updated the marked parts. Could you recheck my changes @tschmidtb51 ?

**Review by tschmidtb51 — APPROVED** [review]
LGTM. Thank you for the changes.

**Pull request merged**